File: samples/conference/registration-bc/src/main/java/com/microsoft/conference/registration/readmodel/QueryServices/IConferenceQueryService.cs
using System;
using System.Collections.Generic;

namespace Registration.ReadModel
{
    public interface IConferenceQueryService
    {
        ConferenceDetails GetConferenceDetails(string slug);
        ConferenceAlias GetConferenceAlias(string slug);

        IList<ConferenceAlias> GetPublishedConferences();
        IList<SeatType> GetPublishedSeatTypes(Guid conferenceId);
        IList<SeatTypeName> GetSeatTypeNames(IEnumerable<Guid> seatTypes);
    }
}

File: samples/conference/registration-bc/src/main/java/com/microsoft/conference/registration/readmodel/queryservices/IConferenceQueryService.java
package com.microsoft.conference.registration.readmodel.queryservices;

import java.util.List;

public interface IConferenceQueryService {
    ConferenceDetails getConferenceDetails(String slug);

    ConferenceAlias getConferenceAlias(String slug);

    List<ConferenceAlias> getPublishedConferences();

    List<SeatType> getPublishedSeatTypes(String conferenceId);

    List<SeatTypeName> getSeatTypeNames(List<String> seatTypes);
}
